### DISCUSSION TITLE
Commit amend

### DISCUSSION THREAD

**Discussion by Matsa59** [gitbutlerapp/gitbutler#2821] [Ideas]
It could be really nice to have a checkbox to amend changes to the previous commit

**Comment by schacon** [comment]
You can just drag the files down onto the previous commit. Or shift-select multiple files and drag them down.

![CleanShot 2024-02-23 at 10 10 50](https://github.com/gitbutlerapp/gitbutler/assets/70/579c9e5b-b8d8-4bb0-ba29-1a94fb7112a7)

**Reply by Matsa59** [reply]
oh nice to know, so I guess a small checkbox could be just a small nice to have ;p

**Reply by ryanhiebert** [reply]
I guess I was expecting a button or something? Its easy enough once I know.

**Reply by Byron** [reply]
Now that there is the notion of "Edit Mode" and "Resolve Conflict" mode, I can definitely imagine also having an "Amend" mode to facilitate using a more Git-like workflow. Is this what you had in mind?

**Reply by Caleb-T-Owens** [reply]
@Byron I'm curious, how would an "Amend" mode work?

**Reply by Byron** [reply]
Thinking about it… I think it's "Edit commit" 😄. So it does exist already.

**Reply by mkotsbak** [reply]
No, its not. Edit commit requires you to enable it before making the modifications. Now I did them first and want something similar to "git commit --amend".

**Reply by Byron** [reply]
You should be able to drag the changes into the commit that they should be amended to.

**Comment by Caleb-T-Owens** [comment]
Aside from perhaps better documentation or onboarding, this seems resolved, so I'll close this discussion for now. Feel free to reopen if anyone has better ideas of how an amending workflow could work. We always appreciate new ideas.

Thank you to all for participating

**Reply by Caleb-T-Owens** [reply]
As an internet boomer, I really have no idea how the statuses of discussions work, so I think I've closed it 😆

**Reply by mkotsbak** [reply]
I dont like being forced to use drag and drop for a feature. I was looking at the "..." on the commit for an option to amend the commit. It would open a commit box with a checkbox to enable amend and displays old commit message. Like in IntelliJ commit box.
